### DISCUSSION TITLE
[Feature] Environment variables for GitOps automation

### DISCUSSION THREAD

**Discussion by CygnusHyoga** [immich-app/immich#4144] [Feature Request]
### The feature

Hello! It would be a good idea to add environment variables, to be able to automate the end-to-end installation of immich server.

For example, a typical case, when it is installed by GitOps (ArgoCD, Flux, etc.), it would be ideal to be able to create the admin user, its password, its token, and its email on the first start of the application without manual user interaction with the GUI. It is quite complex to install the application and then have to make configurations manually, I think it would simplify the installation and testing for many new users.

Example of environment variables for bootstrap:

IMMICH_BOOTSTRAP_EMAIL: "xxx@example.com"
IMMICH_BOOTSTRAP_PASSWORD: xxxxx
IMMICH_BOOTSTRAP_TOKEN: xxxxxxx

Another thing I have noticed is that the OAUTH environment variables no longer exist, I don't know the reason for removing them, but I think it would be convenient to add them again to be able to configure SSO authentication by code completely.

Thanks! I really like Immich!

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
You can use a config file to configure everything https://immich.app/docs/install/config-file. That would just leave the admin bootstrap, but I think that might then be better covered by allowing to set the admin permission via oauth?

**Comment by CygnusHyoga** [comment]
Thanks for the reply, I didn't know about the config-file, I will try this way for oauth config.

Yes, allowing to set the admin permission via oauth, will be fantastic.
